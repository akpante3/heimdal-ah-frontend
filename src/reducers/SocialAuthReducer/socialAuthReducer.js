// import isEmpty from '../../utils/isEmpty';
// import { ACTIONS } from '../../actions/actionTypes';

// const initialState = {
//   isAuthenticated: false,
//   user: {},
// };

// export default (state = initialState, action) => {
//   switch (action.type) {
//     case ACTIONS.SET_CURRENT_USER:
//       return {
//         ...state,
//         isAuthenticated: !isEmpty(action.payload),
//         user: action.payload,
//       };
//     default:
//       return state;
//   }
// };
