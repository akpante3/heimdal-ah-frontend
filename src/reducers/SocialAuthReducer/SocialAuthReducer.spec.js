/* eslint-disable no-undef */
import SET_CURRENT_USER from '../../actions/actionTypes';

// Reducer to be tested
import socialAuthReducer from './socialAuthReducer';

describe('select_reducer', () => {
  it('testing our initial state', () => {
    const action = { type: SET_CURRENT_USER };
    const initialState = { isAuthenticated: false, user: {} };
    expect(socialAuthReducer(undefined, action)).toEqual(initialState);
  });

  it('is correct', () => {
    const action = { type: SET_CURRENT_USER };
    expect(socialAuthReducer(undefined, action)).toMatchSnapshot();
  });

  it('returns the correct state', () => {
    const action = { type: SET_CURRENT_USER, payload: 1 };
    expect(socialAuthReducer(undefined, action)).toMatchSnapshot();
  });
});
