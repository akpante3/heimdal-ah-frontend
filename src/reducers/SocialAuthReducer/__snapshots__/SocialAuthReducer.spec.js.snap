// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_reducer is correct 1`] = `
Object {
  "isAuthenticated": false,
  "user": Object {},
}
`;

exports[`select_reducer returns the correct state 1`] = `
Object {
  "isAuthenticated": false,
  "user": Object {},
}
`;
